lib/Basic/FileSystem.cpp

```diff
@@ -62,6 +62,19 @@ class LocalFileSystem : public FileSystem {
     }
     return std::unique_ptr<llvm::MemoryBuffer>(result->release());
   }
+
+  virtual bool remove(const std::string& path) override {
+    struct stat statbuf;
+    if (sys::stat(path.c_str(), &statbuf) != 0) {
```

Comment:
Unfortunately, unlinking a directory just gives `EPERM`, not a distinct error code, so it'll be slightly unclean.